### PR TITLE
MM-56819: Introduce --local mode in MMCTL to handle user preferences.

### PR DESCRIPTION
**Pull request by kaoski**
#### Summary
This commit introduces support for --local mode in MMCTL for managing user preferences. The following changes have been added:
1. API endpoints for user preferences are registered with the local router.
2. Test cases have been added to ensure functionality for both System Admins and --local mode.

Note: As there is no authentication/authorization required in --local mode, @mattermod suggested omitting session checks from the API handlers. But we are already bypassing session checks for local mode requests as detailed [here](https://github.com/mattermost/mattermost/blob/master/server/channels/app/authorization.go#L205). Hence, I have retained the existing session checks in place.

#### Ticket Link
Github Issue: https://github.com/mattermost/mattermost/issues/26864
Jira Ticket: https://mattermost.atlassian.net/browse/MM-56819

#### Release Note
```release-note
Added --local mode support in MMCTL to handle user preferences.'
```

### CODE REVIEW

**Comment by agnivade** [comment]
@hanzei - From what I understood, CRE was looking to own mmctl a bit more. You are most welcome to fully review this by your team. But happy to standby as a secondary if you prefer. LMK.

File: server/channels/api4/preference_test.go [hunk 1]
```diff
@@ -623,6 +662,28 @@ func TestDeletePreferences(t *testing.T) {
 	resp, err = client.DeletePreferences(context.Background(), th.BasicUser.Id, preferences)
 	require.Error(t, err)
 	CheckUnauthorizedStatus(t, resp)
+
+	//DeletePreferences API from System Admin and Local Client
+	th.TestForSystemAdminAndLocal(t, func(t *testing.T, c *model.Client4) {
+
+		//Creating Test Data
+		var preferences model.Preferences
```
Comment (hanzei) [inline server/channels/api4/preference_test.go:670]: CI doesn't like the style
```suggestion
	th.TestForSystemAdminAndLocal(t, func(t *testing.T, c *model.Client4) {
		//Creating Test Data
		var preferences model.Preferences
```

File: server/channels/api4/preference_test.go [hunk 2]
```diff
@@ -255,6 +279,21 @@ func TestUpdatePreferences(t *testing.T) {
 	resp, err = client.UpdatePreferences(context.Background(), user1.Id, preferences1)
 	require.Error(t, err)
 	CheckUnauthorizedStatus(t, resp)
+
+	//UpdatePreferences API from System Admin and Local Client
```
Comment (hanzei) [inline server/channels/api4/preference_test.go:283]: Could you please add a space before each comment as that is the norm in this codebase?

File: server/channels/api4/preference_test.go [hunk 3]
```diff
@@ -623,6 +662,28 @@ func TestDeletePreferences(t *testing.T) {
...
+			Name:     model.NewId(),
+			Value:    "true",
+		}
+		preferences = append(preferences, preference)
+		c.UpdatePreferences(context.Background(), th.BasicUser.Id, preferences)
+
+		//Delete Prefrerences Operation
+		resp, err = c.DeletePreferences(context.Background(), th.BasicUser.Id, preferences)
+
+		//Validations
```
Comment (hanzei) [inline server/channels/api4/preference_test.go:683]: Nit: that is a bit too verbose

```suggestion
		resp, err = c.DeletePreferences(context.Background(), th.BasicUser.Id, preferences)
```

Comment (hanzei) [inline server/channels/api4/preference_test.go:683]: Nit: that is a bit too verbose

```suggestion
		resp, err = c.DeletePreferences(context.Background(), th.BasicUser.Id, preferences)
```

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution @kaoski !

File: server/channels/api4/preference_test.go (hunk 1 above)
Comment (kaoski) [reply server/channels/api4/preference_test.go]: Fixed the CI Issue

File: server/channels/api4/preference_test.go (hunk 2 above)
Comment (kaoski) [reply server/channels/api4/preference_test.go]: Done

File: server/channels/api4/preference_test.go (hunk 3 above)
Comment (kaoski) [reply server/channels/api4/preference_test.go]: Fixed

Comment (kaoski) [reply server/channels/api4/preference_test.go]: Fixed

**Review by hanzei — APPROVED** [review]
LGTM, thanks @kaoski :rocket:

**Comment by hanzei** [comment]
@BenCookie95 The PR is ready for your review

**Comment by hanzei** [comment]
CI error seems unrelated. Restarting the job run

**Comment by kaoski** [comment]
@hanzei @BenCookie95 E2E smoke tests and enterprise CI/tests are failing, but it's likely not due to this change. Is there anything we can do to resolve this issue?

**Comment by hanzei** [comment]
CI is happy now 😊

**Review by BenCookie95 — APPROVED** [review]
LGTM, thank you!

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Cleared MMCTL auth table to ensure this could only work in `-- local` mode
- Test `get`, `set`, `delete` and list features targeting a few settings including some that were not actually set
- Used `list` feature and tested refining with various users and categories  
No issues found.

LGTM!

Huge thanks @kaoski for implementing this in local mode 🎉 
Sorry for the delay in review.

**Comment by cwarnermm** [comment]
@DHaussermann - Are there updates needed to documented mmctl commands based on the changes made in this PR?

**Pull request merged**